src/test/unit/algorithms/ConnectionsTest.cpp

```diff
@@ -480,6 +481,143 @@ TEST(ConnectionsTest, testRaisePermanencesToThresholdOutOfBounds) {
     << "raisePermanence fails when lower number of available synapses than requested by threshold";
 }
 
+TEST(ConnectionsTest, testSynapseCompetition) {
+
+  struct testCase {
+    UInt nsyn; // Total number of potential synapses on segment
+    UInt ncon; // Number of connected synapses, before calling synapseCompetition
+    UInt min;  // Bounds of synapseCompetition
+    UInt max;  // Bounds of synapseCompetition
+    // The target number of synapses can't be met, just make sure it does not crash.
+    bool dont_check = false;
+  };
+
+  testCase emptySegment;
+  emptySegment.nsyn = 0;
+  emptySegment.ncon = 0;
+  emptySegment.min  = 3;
+  emptySegment.max  = 100;
+  emptySegment.dont_check = true;
```

Comment:
can we create a new synapse on an empty seg, if needed, rather than failing?

Reply:
The synapse competition method does not have access to the current inputs. The call would need to specify which presynapses to use.